qa/L0_backend_python/argument_validation/models/argument_validation/1/model.py

```diff
@@ -0,0 +1,132 @@
+# Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#  * Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+#  * Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#  * Neither the name of NVIDIA CORPORATION nor the names of its
+#    contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import numpy as np
+import unittest
+import triton_python_backend_utils as pb_utils
+
+
+class ArgumentValidationTest(unittest.TestCase):
+    def test_infer_request_args(self):
+        # Dummy arguments used in the tests.
+        inputs = [pb_utils.Tensor('INPUT0', np.asarray([1, 2], dtype=np.int32))]
+        model_name = 'my_model'
+        requested_output_names = ['my_output']
+
+        # inputs field validation
+        with self.assertRaises(BaseException) as e:
+            pb_utils.InferenceRequest(inputs=[None], model_name=model_name,
+                    requested_output_names=requested_output_names)
+
+        with self.assertRaises(BaseException) as e:
```

Comment:
Does error message checking not matter here? If there's an easy way to add that, it could be useful. That'd be more specific, so the exception isn't being returned for the wrong reason and we fail to actually check something.

If it'd add testing complexity, it could be not worth it, since the test is checking the desired behavior.

Reply:
I would prefer a more specific error checking in this case even if it adds test complexity. The correct error messages are crucial for developer. 
Also, checking for specific error message would make the documentation of these test you requested above redundant.

Reply:
Is `with self.assertRaises(BaseException) as e:` similar to `try ... catch ...` that expects an exception to be raised? If we are checking exception, shouldn't we expect a more specific exception instead of `BaseException`? i.e. `InferenceServerException` in client testing.

Reply:
> Is with self.assertRaises(BaseException) as e: similar to try ... catch ... that expects an exception to be raised? If we are checking exception, shouldn't we expect a more specific exception instead of BaseException? i.e. InferenceServerException in client testing.

@GuanLuo I'll make this change.

@dyastremsky @tanmayv25 The errors are quite verbose so I'd rather not check for the exact error message here. I think this will make the test bloated with repetitive messages. I've shown an example below:

```
TypeError: __init__(): incompatible constructor arguments. The following argument types are supported:
    1. c_python_backend_utils.InferenceRequest(request_id: str = '', correlation_id: int = 0, inputs: List[triton::backend::python::PbTensor], requested_output_names: List[str], model_name: str, model_version: int = -1, flags: int = 0)

Invoked with: kwargs: inputs=None, model_name='my_model', requested_output_names=['my_output']
```

Reply:
Makes sense. The tests are clear and the specific error checking is good, so approving. For future PRs, we could probably do a self.assertIn(str(e.exception), "incompatible constructor arguments").

Alternatively, if checking many error messages like in this test, it may be worthwhile to write a function like [this one.](https://stackoverflow.com/a/8673096), except with assertIn. Then, you can just add a few words from the error as a parameter for assertRaisesWithMessage.

Reply:
> Alternatively, if checking many error messages like in this test, it may be worthwhile to write a function like [this one.](https://stackoverflow.com/a/8673096), except with assertIn. Then, you can just add a few words from the error as a parameter for assertRaisesWithMessage.

Sounds like a good suggestion that can apply to many of our unittests. Can you file a ticket for this?

Reply:
Sure, see here: https://jirasw.nvidia.com/browse/DLIS-3689